awswrangler/opensearch/_utils.py

```diff
@@ -37,13 +55,105 @@ def _is_https(port: int) -> bool:
+    return policy
+
+
+def _get_default_network_policy(collection_name: str, vpc_endpoints: Optional[List[str]]) -> List[Dict[str, Any]]:
+    policy: List[Dict[str, Any]] = [
+        {
+            "Rules": [
+                {
+                    "ResourceType": "dashboard",
+                    "Resource": [
+                        f"collection/{collection_name}",
+                    ],
+                },
+                {
+                    "ResourceType": "collection",
+                    "Resource": [
+                        f"collection/{collection_name}",
+                    ],
+                },
+            ],
+            "Description": f"Default network policy for collection '{collection_name}'.",
+        }
+    ]
+    if vpc_endpoints:
+        policy[0]["SourceVPCEs"] = vpc_endpoints
+    else:
+        policy[0]["AllowFromPublic"] = True
+    return policy
+
+
+def _create_security_policy(
+    collection_name: str,
+    policy: Optional[Union[Dict[str, Any], List[Dict[str, Any]]]],
+    policy_type: str,
+    client: boto3.client,
+    **kwargs: Any,
+) -> None:
+    if not kwargs:
+        kwargs = {}
+    if policy_type == "encryption" and not policy:
+        policy = _get_default_encryption_policy(collection_name, kwargs.get("kms_key_arn"))
+    elif policy_type == "network" and not policy:
+        policy = _get_default_network_policy(collection_name, kwargs.get("vpc_endpoints"))
+    else:
+        raise exceptions.InvalidArgument(f"Invalid policy type '{policy_type}'.")
+    try:
+        client.create_security_policy(
+            name=f"{collection_name}-{policy_type}-policy",
+            policy=json.dumps(policy),
+            type=policy_type,
+            description=f"Default {policy_type} policy for collection '{collection_name}'.",
+        )
+    except botocore.exceptions.ClientError as error:
+        if error.response["Error"]["Code"] == "ConflictException":
+            raise exceptions.Conflict("The policy name or rules conflict with an existing policy.") from error
+        raise error
+
+
+def _create_data_policy(collection_name: str, policy: List[Dict[str, Any]], client: boto3.client) -> None:
+    try:
+        client.create_access_policy(
+            name=f"{collection_name}-data-policy",
+            policy=json.dumps(policy),
+            type="data",
+            description=f"Default data policy for collection '{collection_name}'.",
+        )
+    except botocore.exceptions.ClientError as error:
+        if error.response["Error"]["Code"] == "ConflictException":
+            raise exceptions.Conflict("The policy name or rules conflict with an existing policy.") from error
+        raise error
+
+
 def connect(
     host: str,
     port: Optional[int] = 443,
     boto3_session: Optional[boto3.Session] = None,
     region: Optional[str] = None,
     username: Optional[str] = None,
     password: Optional[str] = None,
+    service: Optional[str] = None,
+    timeout: int = 30,
```

Comment:
I'm passing these as args with the defaults without change for now, but we might want to review them. For example, 10 retries seems like too much